src/engine/prompt.go

```diff
@@ -40,6 +46,10 @@ func (e *Engine) Primary() string {
 		e.write(" ")
 	}
 
+	if e.Config.ShellIntegration {
```

Comment:
The problem with WT and multiline transient prompt can be avoided by not printing `FTCS_COMMAND_START` at the end of primary prompt if transient prompt is enabled. In VS Code, iTerm2 and kitty it works like before if `FTCS_COMMAND_START` is sent only when drawing the transient prompt.

<details>
<summary>Logpoints from WT:</summary>

```
// start - drawing primary prompt
end: x = 0, y = 2
prompt: x = 0, y = 2
command: x = 3, y = 4
// echo "a" ENTER, drawing transient prompt over primary
command: x = 3, y = 3
output: x = 0, y = 4
// a, drawing primary prompt
end: x = 0, y = 5
prompt: x = 0, y = 5
command: x = 3, y = 7
// echo "b" ENTER, drawing transient prompt over primary
command: x = 3, y = 6
output: x = 0, y = 7
// b, drawing primary prompt
end: x = 0, y = 8
prompt: x = 0, y = 8
command: x = 3, y = 10
```

</details>

```suggestion
	if e.Config.ShellIntegration && e.Config.TransientPrompt == nil {
```